File: C++/Basic/Vector/Vector.cpp
Vector::Vector(double x=0, double y=0) {
        this->x = x;
        this->y = y;
}

Vector::~Vector() {}

double Vector::getX() const {
        return this->x;
}

double Vector::getY() const {
        return this->y;
}

void Vector::setX(double value) {
        this->x = value;
}

void Vector::setY(double value) {
        this->y = value;
}

double Vector::len() const {
        return hypot(this->x, this->y);
}

bool Vector::operator==(const Vector& other) const {
        return this->x == other.x && this->y == other.y;
}

bool Vector::operator!=(const Vector& other) const {
        return !(*this == other);
}

void Vector::operator+=(const Vector& other) {
        this->x += other.x;
        this->y += other.y;
}

void Vector::operator-=(const Vector& other) {
        this->x -= other.x;
        this->y -= other.y;
}

Vector Vector::operator+(const Vector& other) const {
        Vector sum = Vector(this->x+other.x, this->y+other.y);
        return sum;
}

Vector Vector::operator-(const Vector& other) const {
        Vector diff = Vector(this->x-other.x, this->y-other.y);
        return diff;
}

std::ostream& operator<<(std::ostream& out, const Vector& vector) {
        out << '(' << v.getX() << ", " << v.getY() << ')';
        return out;
}

File: Java/Basic/Vector/Vector.java
class Vector {
	private double x;
	private double y;

	Vector::Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

	~Vector() {}

	public double getX() {
		return x;
	}

	public double getY() {
		return y;
	}

	public void setX(double value) {
		x = value;
	}

	public void setY(double value) {
		y = value;
	}

	public double len() {
		return Math.hypot(x, y);
	}

	public boolean equals(Vector other) {
		return x == other.x && y == other.y;
	}

	public boolean notequals(Vector other) {
        return !(*this == other);
    }

    public void increment(Vector other) {
        x += other.x;
        y += other.y;
    }

    public void decrement(Vector other) {
        x -= other.x;
        y -= other.y;
    }

    public Vector summary( Vector other) {
        Vector sum = Vector(x+other.x, y+other.y);
        return sum;
    }

    public Vector substraction( Vector other) {
        Vector diff = Vector(x-other.x, y-other.y);
        return diff;
    }

    std::ostream& operator<<(std::ostream& out, Vector& vector) {
        out << '(' << v.getX() << ", " << v.getY() << ')';
        return out;
    }
}